src/zenml/model_registries/base_model_registry.py

```diff
@@ -0,0 +1,425 @@
+        self,
+        name: str,
+        description: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ) -> ModelRegistration:
+        """Registers a model in the model registry.
+
+        Args:
+            name: The name of the registered model.
+            description: The description of the registered model.
+            tags: The tags associated with the registered model.
+        """
+
+    @abstractmethod
+    def delete_model(
+        self,
+        name: str,
+    ) -> None:
+        """Deletes a registered model from the model registry.
+
+        Args:
+            name: The name of the registered model.
+        """
+
+    @abstractmethod
+    def update_model(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ) -> ModelRegistration:
+        """Updates a registered model in the model registry.
+
+        Args:
+            name: The name of the registered model.
+            description: The description of the registered model.
+            tags: The tags associated with the registered model.
+        """
+
+    @abstractmethod
+    def get_model(self, name: str) -> ModelRegistration:
+        """Gets a registered model from the model registry.
+
+        Args:
+            name: The name of the registered model.
+
+        Returns:
+            The registered model.
+        """
+
+    @abstractmethod
+    def list_models(
+        self,
+        name: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ) -> List[ModelRegistration]:
+        """Lists all registered models in the model registry.
+
+        Args:
+            name: The name of the registered model.
+            tags: The tags associated with the registered model.
+
+        Returns:
+            A list of registered models.
+        """
+
+    @abstractmethod
+    def check_model_exists(self, name: str) -> bool:
+        """Checks if a model exists in the model registry.
+
+        Args:
+            name: The name of the registered model.
+
+        Returns:
+            True if the model exists, False otherwise.
+        """
+
+    # ---------
+    # Model Version Methods
+    # ---------
+
+    @abstractmethod
+    def register_model_version(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+        model_source_uri: Optional[str] = None,
+        version: Optional[str] = None,
+        version_description: Optional[str] = None,
+        version_tags: Optional[Dict[str, str]] = None,
+        registry_metadata: Optional[Dict[str, str]] = None,
+        zenml_version: Optional[str] = None,
+        zenml_pipeline_run_id: Optional[str] = None,
+        zenml_pipeline_name: Optional[str] = None,
+        zenml_step_name: Optional[str] = None,
+        **kwargs: Any,
+    ) -> ModelVersion:
+        """Registers a model version in the model registry.
+
```

Comment:
You should set out some rules in the abstraction concerning what happens if you already have a model with that same name in the registry, or even a model with the same version. Do you expect a `EntityExistsError` to be raised ? What if you don't supply a version ? Do you expect the implementation to apply some kind of auto-increment logic to generate a new version ?

Or maybe the implementation doesn't even support specifying a version, but always implements auto-increment for versioning.